packages/blockchain-wallet-v4-frontend/src/scenes/BuySell/ExchangeCheckout/index.js

```diff
@@ -82,8 +82,8 @@ class ExchangeCheckout extends React.PureComponent {
     this.state = {}
   }
 
-  componentWillReceiveProps (nextProps) {
-    const { base, quote } = nextProps
+  componentDidUpdate (prevProps) {
+    const { base, quote } = prevProps
 
     if (quote && !equals(this.props.quote, quote)) {
       if (base === 'fiat') {
```

Comment:
There's setState down the code
this will lead to infinite update loop

Reply:
math should be done in selectors to fix that

Reply:
Should I refactor the code/math into one of the selectors.js files?